static/js/filter.js

```diff
@@ -63,8 +64,11 @@ function message_in_home(message) {
         return true;
     }
 
-    // We don't display muted streams in 'All messages' view
-    return !stream_data.is_muted(message.stream_id);
+    // We don't display muted streams and topics in 'All messages' view
+    return (
+        !stream_data.is_muted(message.stream_id) ||
+        muted_topics.is_topic_muted(message.stream_id, message.topic)
```

Comment:
Ahh, I have written this condition w.r.t. muted topic (should be negated).
Fixing it.